### PR TITLE
Updates demo dependencies. Minor code tweaks

### PR DESCRIPTION
**Pull request by cjbj**
Signed-off-by: Christopher Jones <christopher.jones@oracle.com>

### CODE REVIEW

File: javascript/rest-api/part-2-database-basics/hr_app/services/database.js [hunk 1]
```diff
@@ -1,43 +1,47 @@
 const oracledb = require('oracledb');
 const dbConfig = require('../config/database.js');
 
+// Tell node-oracledb where to find the Oracle Instant Client 'Basic' package on macOS and Windows
+if (process.platform === 'darwin') {
+  oracledb.initOracleClient({libDir: process.env.HOME + '/Downloads/instantclient_19_8'});
+} else if (process.platform === 'win32') {
+  oracledb.initOracleClient({libDir: 'C:\\oracle\\instantclient_19_8'});   // note the double backslashes
+} // else you must set the system library search path before starting Node.js
```
Comment (dmcghan) [inline javascript/rest-api/part-2-database-basics/hr_app/services/database.js:9]: Good stuff!

File: javascript/rest-api/part-2-database-basics/hr_app/services/database.js [hunk 2]
```diff
@@ -1,43 +1,47 @@
...
+  } finally {
+    if (conn) { // conn assignment worked, need to close
+      try {
+        await conn.close();
+      } catch (err) {
+        console.error(err);
       }
     }
-  });
+  }
```
Comment (dmcghan) [inline javascript/rest-api/part-2-database-basics/hr_app/services/database.js:44]: I like this better than using an explicit promise.

File: javascript/rest-api/part-2-database-basics/hr_app/package.json [hunk 3]
```diff
@@ -9,8 +9,8 @@
   "keywords": [],
   "license": "Apache-2.0",
   "dependencies": {
-    "express": "^4.16.4",
-    "morgan": "^1.9.1",
-    "oracledb": "^3.1.2"
+    "express": "^4.17.1",
+    "morgan": "^1.10.0",
+    "oracledb": "^5.1.0"
```
Comment (dmcghan) [inline javascript/rest-api/part-2-database-basics/hr_app/package.json:14]: Nice! :)

**Review by dmcghan — APPROVED** [review]
Excellent changes. Thanks, Chris!

**Review by anthony-tuininga — APPROVED** [review]

**Pull request merged**